Examples/LambdaFunctions/Sources/LambdaAPI/ProductLambdaHandler.swift

```diff
@@ -0,0 +1,222 @@
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+import AWSLambdaRuntime
+import NIO
+import Logging
+import AWSLambdaEvents
+
+struct ProductLambdaHandler: EventLoopLambdaHandler {
+    
+    typealias In = APIGateway.V2.Request
+    typealias Out = APIGateway.V2.Response
+    
+    let service: ProductService
+    let operation: Operation
+    
+    func handle(context: Lambda.Context, event: APIGateway.V2.Request) -> EventLoopFuture<APIGateway.V2.Response> {
+    
+        switch operation {
+        case .create:
+            logger.info("create")
+            let create = CreateLambdaHandler(service: service).handle(context: context, event: event)
+                .flatMap { response -> EventLoopFuture<APIGateway.V2.Response> in
+                switch response {
+                case .success(let result):
+                    let value = APIGateway.V2.Response(with: result, statusCode: .created)
+                    return context.eventLoop.makeSucceededFuture(value)
+                case .failure(let error):
+                    let value = APIGateway.V2.Response(with: error, statusCode: .forbidden)
+                    return context.eventLoop.makeSucceededFuture(value)
+                }
+            }
+            return create
+        case .read:
+            logger.info("read")
+            let read = ReadLambdaHandler(service: service).handle(context: context, event: event)
+                .flatMap { response -> EventLoopFuture<APIGateway.V2.Response> in
+                switch response {
+                case .success(let result):
+                    let value = APIGateway.V2.Response(with: result, statusCode: .ok)
+                    return context.eventLoop.makeSucceededFuture(value)
+                case .failure(let error):
+                    let value = APIGateway.V2.Response(with: error, statusCode: .notFound)
+                    return context.eventLoop.makeSucceededFuture(value)
+                }
+            }
+            return read
+        case .update:
+            logger.info("update")
+            let update = UpdateLambdaHandler(service: service).handle(context: context, event: event)
+                .flatMap { response -> EventLoopFuture<APIGateway.V2.Response> in
+                switch response {
+                case .success(let result):
+                    let value = APIGateway.V2.Response(with: result, statusCode: .ok)
+                    return context.eventLoop.makeSucceededFuture(value)
+                case .failure(let error):
+                    let value = APIGateway.V2.Response(with: error, statusCode: .notFound)
```

Comment:
ditto on error handling and response code

Reply:
Refactored in 8649522